pkg/controllers/spannerautoscaler_controller.go

```diff
@@ -322,23 +336,36 @@ func (r *SpannerAutoscalerReconciler) needUpdateNodes(sa *spannerv1alpha1.Spanne
 	}
 }
 
+// For testing purpose only
 func calcDesiredNodes(currentCPU, currentNodes, targetCPU, minNodes, maxNodes, maxScaleDownNodes int32) int32 {
-	totalCPU := currentCPU * currentNodes
-	desiredNodes := totalCPU/targetCPU + 1 // roundup
+	return calcDesiredProcessingUnits(currentCPU, currentNodes*1000, targetCPU, minNodes*1000, maxNodes*1000, maxScaleDownNodes) / 1000
+}
+
+func roundUpProcessingUnits(processingUnits int32) int32 {
+	if processingUnits >= 1000 {
+		return ((processingUnits / 1000) + 1) * 1000
+	}
```

Comment:
Is this really required? I think we should just keep on increasing in multiple of 100 processing units instead of adding new full nodes.
For example, suppose current processing units value is 1800 and the calculated desired value turns out to be 2240. Then we should round up to 2300 instead of rounding up to 3000.

Let me know your thoughts.

Reply:
Sorry, I just re-read the docs and I now understand that it is not possible to use increments of 100 after 1000. Please ignore this.

Reply:
> Just changing processingUnits >= 1000 to processingUnits > 1000 should be sufficient in that case I think.

It will not work as intended because of `roundUpProcessingUnits(1000) = 1100`.

```go
package main

import (
	"fmt"
)

func roundUpProcessingUnits(processingUnits int32) int32 {
	if processingUnits > 1000 {
		return ((processingUnits / 1000) + 1) * 1000
	}
	return ((processingUnits / 100) + 1) * 100
}

func main() {
	for _, target := range []int32{100, 900, 1000, 2000} {
		fmt.Printf("roundUpProcessingUnits(%d)=%d\n", target, roundUpProcessingUnits(target))
	}
}
```

```
roundUpProcessingUnits(100)=200
roundUpProcessingUnits(900)=1000
roundUpProcessingUnits(1000)=1100
roundUpProcessingUnits(2000)=3000
```

Reply:
I think it is better to add a unit test for `roundUpProcessingUnits` function.

Reply:
Add the unit test in https://github.com/mercari/spanner-autoscaler/pull/34/commits/1b30cf264ab93624f8fe80dd5d7ce0b2a78e3b79

Reply:
Thank you for the explanation, I got it now!